src/lib.rs

```diff
@@ -92,6 +92,33 @@ mod lint_tests {
     assert_diagnostic(&diagnostics[1], "ban-unknown-rule-code", 4, 3, src);
   }
 
+  #[test]
+  fn no_ban_unknown_rule_code_error_without_rule() {
```

Comment:
This test name sounds a little bit unclear to me. How about `dont_run_ban_unknown_rule_code_when_no_rules_provided` or something?